README.rst

```diff
@@ -1,7 +1,108 @@
-more generally. It is currently WIP.
+This package provides a plugin for the `pytest`_ framework that allows
+developers to control unit tests that require access to data from the internet.
+It was originally part of the `astropy`_ core package, but has been moved to a
+separate package in order to be of more general use.
+
+.. _pytest: https://pytest.org/en/latest/
+.. _astropy: https://astropy.org/en/latest/
+
+
+Motivation
+----------
+
+Many software packages provide features that require access to data from the
+internet. These features need to be tested, but unit tests that access the
+internet can dominate the overall runtime of a test suite.
+
+The ``pytest-remotedata`` plugin allows developers to indicate which unit tests
+require access to the internet, and to control when and whether such tests
+should execute as part of any given run of the test suite.
+
+Installation
+------------
+
+The ``pytest-remotedata`` plugin can be installed using ``pip``::
+
+    $ pip install pytest-remotedata
+
+It can also be installed from source::
+
+    $ git clone https://github.com/astropy/pytest-remotedata
+    $ cd pytest-remotedata
+    $ python ./setup.py install
+
+In either case, the plugin will automatically be registered for use with
+``pytest``.
+
+Usage
+-----
+
+Installing this plugin makes two decorators available for use with ``pytest``:
+
+* ``remote_data`` for marking tests that require data from the internet
+* ``internet_off`` for marking tests that should run only when internet access
+  is disabled
+
+These decorators can be used to mark test functions, methods, and classes using
+``@pytest.mark``. For example, consider the following test function that
+requires access to data from the internet:
+
+.. code-block:: python
+
+    import pytest
+    from urllib.request import urlopen
+
+    @pytest.mark.remote_data
+    def test_remote_data():
+        urlopen('https://astropy.org')
+
+Marking the ``test_remote_data`` function with ``@pytest.mark.remote_data``
+indicates to ``pytest`` that this test should be run only when access to remote
+data sources is explicitly requested.
+
+When this plugin is installed, the ``--remote-data`` command line option is
+added to the ``pytest`` command line interface.
+
+The default behavior is to skip tests that are marked with ``remote_data``.
+If the ``--remote-data`` option is not provided to the ``pytest`` command, or
+if ``--remote-data=none`` is provided, all tests that are marked with
+``remote_data`` will be skipped. All tests that are marked with
+``internet_off`` will be executed. Any test that attempts to access the
+internet but is not marked with ``remote_data`` will fail.
+
+Providing either the ``--remote-data`` option, or ``--remote-data=any`` to the
+``pytest`` command line interface will cause all tests that are marked with
+``remote-data`` to execute. Any tests that are marked with ``internet_off``
+will be skipped.
```

Comment:
Ah I misunderstood the purpose of this decorator - I thought it would always run tests and temporarily disable internet on a test by test basis.

Reply:
That would be a much more useful feature, wouldn't it?

Reply:
No :) Anyway ignore comment, was just thinking out loud

Reply:
Haha, well you already convinced me actually, so I opened an issue.